tests/cancun/eip4788_beacon_root/common.py

```diff
@@ -0,0 +1,68 @@
+"""
+Common constants, classes & functions local to EIP-4788 tests.
+"""
+
+from ethereum_test_tools import (
+    BeaconRoot,
+    Storage,
+)
+
+REF_SPEC_4788_GIT_PATH = "EIPS/eip-4788.md"
+REF_SPEC_4788_VERSION = "f0eb6a364aaf5ccb43516fa2c269a54fb881ecfd"
+
+BEACON_ROOT_PRECOMPILE_ADDRESS = 0x0B  # HISTORY_STORE_ADDRESS
+BEACON_ROOT_PRECOMPILE_GAS = 4_200  # G_BEACON_ROOT
+HISTORICAL_ROOTS_MODULUS = 98_304
+
+FORK_TIMESTAMP = 15_000  # ShanghaiToCancun timestamp
+DEFAULT_BEACON_ROOT_HASH = BeaconRoot
+
+
+def timestamp_index(timestamp: int) -> int:
+    """
+    Derive the timestamp index into the timestamp ring buffer.
+    """
+    return timestamp % HISTORICAL_ROOTS_MODULUS
+
+
+def root_index(timestamp: int) -> int:
+    """
+    Derive the root index into the root ring buffer.
+    """
+    return timestamp_index(timestamp) + HISTORICAL_ROOTS_MODULUS
+
+
+def expected_storage(
+    beacon_root: bytes,
+    timestamp: int,
+    valid_call: bool,
+    valid_input: bool,
+) -> Storage:
+    """
+    Derives the expected storage for a given beacon root precompile call
+    dependent on:
+    - success or failure of the call
+    - validity of the timestamp input used within the call
+    """
+    storage: Storage.StorageDictType = dict()
```

Comment:
Could we init a `Storage` type directly here? I started using that and it's a bit cleaner than defining a dict, and it has the same characteristics:
```
storage = Storage()
```